java/core/src/main/java/sleeper/core/partition/PartitionTree.java

```diff
@@ -93,6 +93,10 @@ public List<Partition> getAllPartitions() {
         return List.copyOf(idToPartition.values());
     }
 
+    public List<Partition> getAllLeafPartitions() {
+        return List.copyOf(idToPartition.values().stream().filter(Partition::isLeafPartition).collect(Collectors.toList()));
```

Comment:
List.copyOf isn't needed, since Collectors.toList creates a new list.

Reply:
This is also untested production code that was added to use in a test assertion.

It's probably best to be explicit about which partitions should be there in an assertion, rather than rely on this. This method should probably be removed from the production code since it's only used in tests.